compiler/src/lnntoamm/Expr.ts

```diff
@@ -493,9 +497,9 @@ class Call extends Expr {
   */
   inline(amm: Output, kind: AssignKind, name: string, ty: Builtin) {
     const argTys = this.args.map(arg => arg.ty.instance());
-    const selected = this.fns.reverse().find(fn => fn.acceptsTypes(argTys)) || null;
+    const selected = Fn.select(this.fns.reverse(), argTys, this.scope) || [];
```

Comment:
Should the `reverse` be part of calling `Fn.select` or should this API have this explicitly exposed? (Are there any situations where you wouldn't reverse the list, basically, and therefore the `select` function should be taking care of this to DRY it out?)

Reply:
actually, that `reverse` shouldn't be there - the last-defined functions are already at the end of the list, and we `pop` to get the function we use. The bdd tests probably didn't catch this since the only scenario where the reversed list has more than 1 element is when looking at the integer types (and all integers are compatible once compiled).

fixing!